### PR TITLE
Add method to access the materialLawManager

### PR DESCRIPTION
**Pull request by totto82**

### CODE REVIEW

File: opm/core/props/satfunc/SaturationPropsFromDeck.hpp [hunk 1]
```diff
@@ -129,6 +129,9 @@ namespace Opm
                              const double pcow, 
                              double & swat);
 
+        std::shared_ptr<const MaterialLawManager> materialLawManager() const { return materialLawManager_; }
```
Comment (atgeirr) [inline opm/core/props/satfunc/SaturationPropsFromDeck.hpp]: This method should be documented, and rather return a `const MaterialLawManager&`.

Comment (totto82) [inline opm/core/props/satfunc/SaturationPropsFromDeck.hpp]: If I understood you correct. You want: 
```
const std::shared_ptr<const MaterialLawManager>& materialLawManager() const { return materialLawManager_; }
```
Then I get "warning: returning reference to temporary [-Wreturn-local-addr]"

The warning disappear when I change it back to how it was before I changed it according to @andlaus comment. 
```
const std::shared_ptr<MaterialLawManager>& materialLawManager() const { return materialLawManager_; }
```

Comment (atgeirr) [inline opm/core/props/satfunc/SaturationPropsFromDeck.hpp]: No, what I want is
```
const MaterialLawManager& materialLawManager() const { return *materialLawManager_; }
```

**Comment by atgeirr** [comment]
Thanks, merging.

**Pull request merged**